lib/pbench/server/auth/auth.py

```diff
@@ -1,25 +1,80 @@
+from configparser import NoOptionError
 import datetime
 from http import HTTPStatus
 import os
-from typing import Optional
+from typing import List, Optional
 
 from flask import abort, request
 from flask_httpauth import HTTPTokenAuth
 import jwt
 
+from pbench.server import PbenchServerConfig
 from pbench.server.auth import OpenIDClient, OpenIDClientError
 from pbench.server.database.models.active_tokens import ActiveTokens
 from pbench.server.database.models.users import User
 
 
+class InternalUser:
+    """
+    Internal user class for storing user related fields fetch
+    from OIDC token decode.
+    """
+
+    def __init__(
+        self,
+        id: str,
+        username: str,
+        email: str,
+        first_name: str = None,
+        last_name: str = None,
+        roles: List[str] = None,
+    ):
+        self.id = id
+        self.username = username
+        self.email = email
+        self.first_name = first_name
+        self.last_name = last_name
+        self.roles = roles
+
+    def __str__(self):
+        return f"User, id: {self.id}, username: {self.username}"
+
+    def is_admin(self):
+        return "ADMIN" in self.roles
+
+
 class Auth:
     token_auth = HTTPTokenAuth("Bearer")
+    oidc_client: OpenIDClient = None
 
     @staticmethod
     def set_logger(logger):
         # Logger gets set at the time of auth module initialization
         Auth.logger = logger
 
+    @staticmethod
+    def set_oidc_client(server_config: PbenchServerConfig):
+        """
+        OIDC client initialization for third party token verification
+        Args:
+            server_config: Parsed Pbench server configuration
+        """
+        server_url = server_config.get("authentication", "server_url")
+        client = server_config.get("authentication", "client")
+        realm = server_config.get("authentication", "realm")
+        try:
+            secret = server_config.get("authentication", "secret")
```

Comment:
You should be able to use `fallback=None` and avoid the `NoOptionError` and the whole `try`/`except` here. (Note that the default value for `fallback` is a special `_UNSET` raw object, so `None` will work.)